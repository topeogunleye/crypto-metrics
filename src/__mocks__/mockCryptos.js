export default [
  {
    id: 1,
    name: 'Bitcoin',
    symbol: 'BTC',
    image: 'https://www.cryptocompare.com/media/19633/btc.png',
    current_price: '$12,000.00',
    market_cap: '$1,000,000,000',
    total_volume: '$1,000,000,000',
    change_1h: '$1,000.00',
    change_24h: '$1,000.00',
    change_7d: '$1,000.00',
  },
  {
    id: 2,
    name: 'Ethereum',
    symbol: 'ETH',
    image: 'https://www.cryptocompare.com/media/19632/eth_logo.png',
    current_price: '$12,000.00',
    market_cap: '$1,000,000,000',
    total_volume: '$1,000,000,000',
    change_1h: '$1,000.00',
    change_24h: '$1,000.00',
    change_7d: '$1,000.00',
  },
  {
    id: 3,
    name: 'Litecoin',
    symbol: 'LTC',
    image: 'https://www.cryptocompare.com/media/19782/litecoin-logo.png',
    current_price: '$12,000.00',
    market_cap: '$1,000,000,000',
    total_volume: '$1,000,000,000',
    change_1h: '$1,000.00',
    change_24h: '$1,000.00',
    change_7d: '$1,000.00',
  },
  {
    id: 4,
    name: 'Bitcoin Cash',
    symbol: 'BCH',
    image: 'https://www.cryptocompare.com/media/1383612/bch.png',
    current_price: '$12,000.00',
    market_cap: '$1,000,000,000',
    total_volume: '$1,000,000,000',
    change_1h: '$1,000.00',
    change_24h: '$1,000.00',
    change_7d: '$1,000.00',
  },
  {
    id: 5,
    name: 'Cardano',
    symbol: 'ADA',
    image: 'https://www.cryptocompare.com/media/1383613/ada.png',
    current_price: '$12,000.00',
    market_cap: '$1,000,000,000',
    total_volume: '$1,000,000,000',
    change_1h: '$1,000.00',
    change_24h: '$1,000.00',
    change_7d: '$1,000.00',
  },
];
