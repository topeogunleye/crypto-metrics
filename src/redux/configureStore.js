import thunk from 'redux-thunk';
import { createStore, combineReducers, applyMiddleware } from 'redux';
import { cryptosReducer, cryptoHistoryReducer } from './crypto/crypto';

const reducer = combineReducers({
  cryptosReducer,
  cryptoHistoryReducer,
});

const store = createStore(reducer, applyMiddleware(thunk));

export default store;
