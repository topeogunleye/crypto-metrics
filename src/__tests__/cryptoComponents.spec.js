import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

// mocks
import configureStore from 'redux-mock-store';
import mockCryptos from '../__mocks__/mockCryptos';

const mockStore = configureStore([]);

describe('My Connected React-Redux Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      cryptos: mockCryptos,
    });

    component = renderer.create(
      <Provider store={store}>
        <mockCryptos />
      </Provider>,
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should dispatch an action on page load', () => {
    const actions = store.getActions();
    expect(actions).toEqual([]);
  });
});
