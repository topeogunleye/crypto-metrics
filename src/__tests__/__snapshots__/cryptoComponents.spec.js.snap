// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `<mockCryptos />`;
